src/sentry/tagstore/snuba/backend.py

```diff
@@ -497,6 +497,35 @@ def get_perf_group_list_tag_value(
             for group_id, data in result.items()
         }
 
+    def get_generic_group_list_tag_value(
+        self, project_ids, group_id_list, environment_ids, key, value
+    ):
+        filters = {"project_id": project_ids, "group_id": group_id_list}
+        if environment_ids:
+            filters["environment"] = environment_ids
+        conditions = [[f"tags[{key}]", "=", value]]
```

Comment:
Does this need the `DEFAULT_TYPE_CONDITION`? Also doesn't need to be done in this PR but I noticed the 3 functions `get_group_list_tag_value`, `get_perf_group_list_tag_value` and `get_generic_group_list_tag_value` have a lot of repetition.

Reply:
Nah, I think `DEFAULT_TYPE_CONDITION` was a workaround for data in the errors table when transactions were also stored there. The new dataset won't need that extra filter.

Yeah, I noticed the repetition as well. I took a whack at trying to parameterize it all into common functions for `get_group_list_tag_value` and `get_groups_user_counts`. I left the 'perf' ones alone since those require some special-casing and I didn't wanna extract out too much to the point of incomprehensibility.